Mathlib/LinearAlgebra/Semisimple.lean

```diff
@@ -156,7 +156,12 @@ theorem IsSemisimple.of_mem_adjoin_pair {a : End K M} (ha : a ∈ Algebra.adjoin
   let S := AdjoinRoot ((minpoly K g).map <| algebraMap K R)
   have : Finite K R := (AdjoinRoot.powerBasis' <| minpoly.monic <| isIntegral f).finite
   have : Finite R S := (AdjoinRoot.powerBasis' <| (minpoly.monic <| isIntegral g).map _).finite
-  have : IsScalarTower K R S := .of_algebraMap_eq fun _ ↦ rfl
+  #adaptation_note
+  /--
+  After https://github.com/leanprover/lean4/pull/4119 we either need
+  to specify the `(S := R)` argument, or use `set_option maxSynthPendingDepth 2 in`.
+  -/
+  have : IsScalarTower K R S := .of_algebraMap_eq (S := R) fun _ ↦ rfl
```

Comment:
Using `set_option maxSynthPendingDepth 2 in` instead would make this proof 25% quicker (125K heartbeats instead of the current 166K).

Reply:
Using set_option now.